packages/react/src/auto/polaris/inputs/PolarisBooleanInput.tsx

```diff
@@ -0,0 +1,33 @@
+import type { CheckboxProps } from "@shopify/polaris";
+import { Checkbox } from "@shopify/polaris";
+import React from "react";
+import type { Control } from "react-hook-form";
+import { useController } from "react-hook-form";
+import { useFieldMetadata } from "../../hooks/useFieldMetadata.js";
+
+export const PolarisBooleanInput = (props: { field: string; control?: Control<any> } & Partial<CheckboxProps>) => {
+  const { field: fieldApiIdentifier, control, ...rest } = props;
+
+  const { path, metadata } = useFieldMetadata(fieldApiIdentifier);
+
+  const {
+    field: fieldProps,
+    fieldState: { error },
+  } = useController({
+    control,
+    name: path,
+  });
+
+  return (
+    <Checkbox
+      label={metadata.name}
+      checked={!!fieldProps.value}
+      onChange={fieldProps.onChange}
+      onBlur={fieldProps.onBlur}
+      disabled={fieldProps.disabled}
+      name={fieldProps.name}
```

Comment:
These  can also be {...fieldProps}

`name` also doesn't appear to do anything here `label` takes over